eftcamb/11_EFTCAMB_stability.f90

```diff
@@ -33,211 +33,114 @@ module EFTCAMB_stability
     use EFTCAMB_abstract_model_full
     use EFTCAMB_abstract_model_designer
     use EFTDef
+    use EFTCAMB_cache
 
     implicit none
 
 contains
 
     ! ---------------------------------------------------------------------------------------------
     !> Function that computes if the stability requirements are fullfilled in a given time.
-    function EFTStabilityComputation(a)
+    function EFTStabilityComputation( self, a, eft_par_cache, eft_cache )
 
         implicit none
 
-        real(dl), intent(in) :: a                       !< Input scale factor
+        class(EFTCAMB_model)                         :: self          !< the base class
+        real(dl), intent(in)                         :: a             !< the input scale factor.
+        type(EFTCAMB_parameter_cache), intent(inout) :: eft_par_cache !< the EFTCAMB parameter cache that contains all the physical parameters.
+        type(EFTCAMB_timestep_cache ), intent(inout) :: eft_cache     !< the EFTCAMB timestep cache that contains all the physical values.
         logical              :: EFTStabilityComputation !< Logical value returned by the function. If the model is stable this is True, otherwise False.
 
 
         ! 1) Definitions of variables:
-        logical :: EFT_HaveNan
-        real(dl) :: a2,k,k2,grhob_t,grhor_t,grhoc_t,grhog_t,grhov_t, grho, gpres, EFT_H0
-        real(dl) :: adotoa, adotdota, Hdot, Hdotdot
-        real(dl) :: EFTOmegaV, EFTOmegaP,EFTOmegaPP,EFTOmegaPPP, EFTc, EFTLambda, EFTcdot, EFTLambdadot
-        real(dl) :: EFTGamma1V, EFTGamma1P, EFTGamma2V, EFTGamma2P, EFTGamma3V, EFTGamma3P
-        real(dl) :: EFTGamma4V, EFTGamma4P, EFTGamma5V, EFTGamma5P, EFTGamma6V, EFTGamma6P
-        real(dl) :: EFTgrhoq,EFTgpresq,EFTgrhodotq,EFTgpresdotq
-        real(dl) :: EFTpiA1, EFTpiA2, EFTpiB1, EFTpiB2, EFTpiC, EFTpiD, EFTpiD1, EFTpiD2, EFTAT
-        real(dl) :: EFTtemp_H0,  EFTtemp_Omegac, EFTtemp_Omegab, EFTtemp_Omegav, EFTtemp_TCMB, EFTtemp_nu_massless_degeneracy
-        real(dl) :: EFTtemp_grhom, EFTtemp_grhog, EFTtemp_grhor
+        logical  :: EFT_HaveNan_parameter, EFT_HaveNan_timestep
         real(dl) :: EFT_grhonu, EFT_gpinu, grhormass_t
-        real(dl) :: EFT_grhonu_tot, EFT_gpinu_tot, EFT_gpinudot_tot, grho_matter, gpres_matter
+        real(dl) :: EFT_grhonu_tot, EFT_gpinu_tot, EFT_gpinudot_tot, grho_matter, gpres_matter, EFT_grhonudot, EFT_gpinudot
         real(dl) :: kmax
         real(dl) :: temp1, temp2, temp3, temp4, temp5, tempk
         integer  :: nu_i, ind, ind_max
         real(dl) :: EFT_instability_rate
         real(dl) :: EFT_W0, EFT_W1, EFT_W2, EFT_W3, EFT_W6, EFT_W2P, EFT_W3P, EFT_W6P  ! New stability
```

Comment:
Seems not, in general if a variable is not used feel free to delete it! 
Anyway if you delete something that is useful the code will not compile so it can be easily spotted!